radis/lbl/base.py

```diff
@@ -1586,7 +1613,59 @@ def _calc_noneq_parameters(
             self.calc_einstein_coefficients()
 
         # Caching the non-equilibrium parameters Evib and Erot
-        # First, getting the Electronic States file through partition functions:
+        # Here, we regenerate the cache file if it doesn't exist.
+        # If it exists, then the required parameters have already been added to the dataframe.
+        if cache:
+            for isotope in self._get_isotope_list(molecule):
+                cachefile = DataFileManager(engine).cache_file(
+                    filename.format(molecule=molecule, iso=isotope)
+                )
+                # Generate cache file for later use.
+                # Ignore if cache file already exists
+                if exists(cachefile):
+                    continue
+                else:
+                    # First, getting the Electronic States file through partition functions:
+                    elec_state = self.get_partition_function_calculator(
+                        molecule, isotope, state
+                    ).ElecState
+                    spectroscopic_constant_file = elec_state.jsonfile
+                    # Copying the non-equilibrium parameters into a temporary dataframe for caching
+                    temp_df = df[["Evibl", "Evibu", "Erotl", "Erotu"]].copy()
```

Comment:
why .copy() since you're only going to save it to disk ? (you're doubling memory usage here)